langkit/templates/pkg_common_spec_ada.mako

```diff
@@ -91,6 +91,15 @@ package ${ada_lib_name}.Common is
       % endif
    % endfor
 
+   ## Output a subtype to materialize the set of kinds for synthetic nodes
+   % if ctx.synthetic_nodes:
+      subtype Synthetic_Nodes is ${root_node_kind_name}
+         with Dynamic_Predicate =>
```

Comment:
Thinking about it, having a subtype is good, but having a function (could be a built-in property to simplify implementation and binding) on nodes in addition seems even better.

Thinking about the day where one would be able to treat nodes from LK libraries uniformly, that would allow checking for syntheticness in a language agnostic way.

Reply:
Good point, I’ll add this, thanks!

Reply:
Actually this made me realize that we lacked a is-synthetic predicate in the C/Python APIs, so good catch!